src/StatusesHandler.php

```diff
@@ -0,0 +1,270 @@
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        /* @var $event EventInterface */
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $event->setParams(array_merge(
+            $event->getParams(),
+            $this->_getStatusesParams()
+        ));
+    }
+
+    /**
+     * Attach events for status handling.
+     *
+     * @since [*next-version*]
+     */
+    public function run()
+    {
+        $this->_attach('eddbk_bookings_visible_statuses', function ($event) {
+            $event->setParams([
+                'statuses' => $this->_getVisibleStatuses($this->statuses, $this->hiddenStatuses),
+            ]);
+        });
+
+        /*
+         * Attach status endpoint.
+         */
+        $this->_attach('wp_ajax_set_' . $this->screenOptionsKey, function () {
+            $data = json_decode(file_get_contents('php://input'), true);
+            $statuses = $data['statuses'];
+            $this->_setScreenStatuses($this->screenOptionsKey, $statuses);
+        });
+    }
+
+    /**
+     * Get status related parameters to attach on event.
+     *
+     * @since [*next-version*]
+     *
+     * @return array
+     */
+    protected function _getStatusesParams()
+    {
+        return [
+            /*
+             * All available statuses in application.
+             */
+            'statuses' => $this->_getTranslatedStatuses($this->statuses, $this->statusesLabels),
+
+            /*
+             * Statuses that enabled for filtering bookings.
+             */
+            'screenStatuses' => $this->_getScreenStatuses($this->screenOptionsKey, $this->statuses),
+
+            /*
+             * Endpoint for saving statuses
+             */
+            'statusesEndpoint' => $this->statusesEndpoint,
+        ];
+    }
+
+    /**
+     * Get array of visible statuses for UI.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $statuses       List of all statuses.
+     * @param array $hiddenStatuses List of statuses that shouldn't be shown.
+     *
+     * @return array Resulting array of statuses that should be visible in the UI.
+     */
+    protected function _getVisibleStatuses($statuses, $hiddenStatuses)
+    {
+        return array_values(array_filter($statuses, function ($status) use ($hiddenStatuses) {
+            return !in_array($status, $hiddenStatuses);
+        }));
+    }
+
+    /**
+     * Get all translated statuses.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $statuses       List of statuses
+     * @param array $statusesLabels Map of statuses and it's labels
+     *
+     * @return array Map of statuses codes and translations.
+     */
+    protected function _getTranslatedStatuses($statuses, $statusesLabels)
```

Comment:
Ideally, storage and retrieval of entities such as statuses should be handled by a model. See [card](https://github.com/orgs/RebelCode/projects/2#card-9959796).